test/starlark_tests/verifier_scripts/archive_contents_test.sh

```diff
@@ -133,6 +137,49 @@ if [[ -n "${BINARY_TEST_FILE-}" ]]; then
     fi
   fi
 
+  if [[ -n "${CODESIGN_INFO_CONTAINS-}" || -n "${CODESIGN_INFO_NOT_CONTAINS-}" ]]; then
+    codesign_output="$(mktemp "${TMPDIR:-/tmp}/codesign_output.XXXXXX")"
+    codesign --display --verbose=4 "$path" &> "$codesign_output"
+    if [[ -n "${CODESIGN_INFO_CONTAINS-}" ]]; then
+      for test_codesign_info in "${CODESIGN_INFO_CONTAINS[@]}"
+      do
+        codesign_info_found=false
+        if grep -sq "$test_codesign_info" "$codesign_output"; then
+          codesign_info_found=true
+          break
+        fi
+        if [[ "$codesign_info_found" = false ]]; then
+          fail "Expected codesign info \"$test_codesign_info\" was not found." \
+            "The codesign info in the binary were:$newline"\
+            "$(cat "$codesign_output")"
+        fi
+      done
+    fi
+
+    if [[ -n "${CODESIGN_INFO_NOT_CONTAINS-}" ]]; then
+      for test_codesign_info in "${CODESIGN_INFO_NOT_CONTAINS[@]}"
+      do
+        codesign_info_found=false
+        if grep -sq "$test_codesign_info" "$codesign_output"; then
+          codesign_info_found=true
+          break
+        fi
+        for actual_codesign_info in "${codesign_output[@]}"
+        do
+          if [[ "$actual_codesign_info" == "$test_codesign_info" ]]; then
```

Comment:
should this be less strict, and potentially compare these to `*"$test_codesign_info"*`?